codalab/rest/cli.py

```diff
@@ -126,13 +126,12 @@ def general_command(worksheet_uuid, command):
     if isinstance(command, str):
         # shlex throws ValueError on incorrectly formatted commands
         try:
-            # see https://docs.python.org/2/library/shlex.html#shlex.shlex.escapedquotes
+            # see https://docs.python.org/3.4/library/shlex.html#shlex.shlex.escapedquotes
             # By default, the double quote can be escaped. By setting the
             # escapedquotes property, we are able to escape single quotes as well
             # examples: run '\''
-            lexer = shlex.shlex(command, posix=True)
+            lexer = shlex.shlex(command, posix=True, punctuation_chars=True)
             lexer.escapedquotes = '\'"'
-            lexer.whitespace_split = True
```

Comment:
I removed the whitespace_split=True here due to the following reason: there are two types of chars in command: punctuation chars and whitespace. Punctuation chars should be parsed in the first place and then whitespace. Based on [shlex.whitespace_split](https://docs.python.org/3.4/library/shlex.html#shlex.shlex.whitespace_split), when it sets to True, the command string will just be split by whitespace but not punctuation chars, which can be error-prone.

Reply:
what does this do if the command has other punctuation chars besides `;`? Will this split `python my_script.py` into `python`, `my_script`, `.`, `py`? Is that a problem?

Reply:
These are the examples that I should have added to the original description:
```
stanford@Stanfords-MacBook-Pro deployment % python3
Python 3.7.7 (default, Mar 10 2020, 15:43:33) 
[Clang 11.0.0 (clang-1100.0.33.17)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> import shlex
>>> command = 'cl run --- python myscript.py'
>>> lexer = shlex.shlex(command, posix=True, punctuation_chars=True)
>>> lexer.escapedquotes = '\'"'
>>> list(lexer)
['cl', 'run', '---', 'python', 'myscript.py']
```

Reply:
ah I see. Sounds good then.